test/e2e/provision_libvirt.go

```diff
@@ -0,0 +1,73 @@
+//go:build libvirt
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"libvirt.org/go/libvirt"
+	"os"
+	"os/exec"
+	"path"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+// LibvirtProvisioner implements the CloudProvision interface for Libvirt.
+type LibvirtProvisioner struct {
+	conn    *libvirt.Connect // Libvirt connection
+	network string           // Network name
+	storage string           // Storage pool name
+}
+
+func NewLibvirtProvisioner(network string, storage string) (*LibvirtProvisioner, error) {
+	// TODO: accept a different URI.
+	conn, err := libvirt.NewConnect("qemu:///system")
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Check network and storage are not nil?
+	return &LibvirtProvisioner{
+		conn:    conn,
+		network: network,
+		storage: storage,
+	}, nil
+}
+
+func (l *LibvirtProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
+	cmd := exec.Command("/bin/bash", "-c", "./kcli_cluster.sh create")
+	cmd.Dir = "/home/wmoschet/go/src/github.com/confidential-containers/cloud-api-adaptor/libvirt"
```

Comment:
Is this dir viable?

Reply:
I'm definitely not a go expert, but what about something like:
```
wd, err := os.Getwd()
if err != nil {
      return err
}
cmd.Dir = path.Join(wd, "../../libvirt")
```
?

Reply:
Fixed.